src/test/java/io/debezium/connector/vitess/VitessConnectorIT.java

```diff
@@ -168,6 +168,31 @@ public void shouldReceiveHeartbeatEvents() throws Exception {
         assertThat(records.recordsForTopic(topic).size()).isEqualTo(expectedHeartbeatRecords);
     }
 
+    @Test
+    @FixFor("DBZ-7962")
+    public void shouldReceiveHeartbeatEventsShardedKeyspace() throws Exception {
+        TestHelper.executeDDL("vitess_create_tables.ddl");
+        String expectedShard = "-80";
+        startConnector(config -> config.with(
+                Heartbeat.HEARTBEAT_INTERVAL.name(), 1000).with(
+                        VitessConnectorConfig.SHARD, expectedShard),
+                true);
+        assertConnectorIsRunning();
+
+        String topic = Heartbeat.HEARTBEAT_TOPICS_PREFIX.defaultValueAsString() + "." + TEST_SERVER;
+        int expectedHeartbeatRecords = 1;
+        // Sleep for 3 seconds, heartbeat sent every 1 second
+        Thread.sleep(3 * 1000);
```

Comment:
The tests based on sleep tends to be brittle. Could you please modify it so use `Awaitility` waiting for a message arrival.